src/utils/__tests__/validateENS.test.ts

```diff
@@ -0,0 +1,82 @@
+import validateENS from '../validateENS';
+
+describe('valid names', () => {
+  it('domain', () => {
+    expect(validateENS('lol.eth')).toMatchInlineSnapshot(`
+      Object {
+        "valid": true,
+      }
+    `);
+  });
+
+  it('subdomain', () => {
+    expect(validateENS('super.lol.eth')).toMatchInlineSnapshot(`
+      Object {
+        "valid": true,
+      }
+    `);
+  });
+});
+
+describe('invalid names', () => {
+  it('invalid domain', () => {
+    expect(validateENS('wat')).toMatchInlineSnapshot(`
+      Object {
+        "code": "invalid-domain",
+        "hint": "This is an invalid domain",
+        "valid": false,
+      }
+    `);
+  });
+
+  it('domain with special characters', () => {
+    expect(validateENS('omg$.eth')).toMatchInlineSnapshot(`
+      Object {
+        "code": "invalid-domain-name",
+        "hint": "Your name can not include special characters",
+        "valid": false,
+      }
+    `);
+  });
+
+  it('domain with subdomain fails when `includeSubdomains` is falsy', () => {
+    expect(validateENS('wat.lol.eth', { includeSubdomains: false }))
+      .toMatchInlineSnapshot(`
+      Object {
+        "code": "subdomains-not-supported",
+        "hint": "Subdomains are not supported",
+        "valid": false,
+      }
+    `);
+  });
+
+  it('domain with invalid length', () => {
+    expect(validateENS('no.eth')).toMatchInlineSnapshot(`
+      Object {
+        "code": "invalid-length",
+        "hint": "Your name must be at least 3 characters",
+        "valid": false,
+      }
+    `);
+  });
+
+  it('domain with invalid TLD', () => {
+    expect(validateENS('rofl.lol')).toMatchInlineSnapshot(`
+      Object {
+        "code": "invalid-tld",
+        "hint": "This TLD is not supported",
```

Comment:
maybe change TLD to top level domain? I had to google what TLD means so I'm guessing the average rainbow user also does not know

Reply:
also how is it possible to end up with 'rofl.lol'? isn't '.eth' pegged onto the end of whatever the user inputs by default?

Reply:
yeah it's not possible atm, but this util is more of a high level generic util if we chose to support other TLDs in the future